### PR TITLE
Prevent double counting storage pools

### PR DESCRIPTION
**Pull request by DK101010**
### Description

Currenly, getStoragePoolUsedStats method fetch all pools
independed if a pool have a parent. Therefore as soon
as a datastore cluster is used, the cluster and the storages it contains are calculated together
The change fetch all storages without cluster and all datastore cluster

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

tested in own env.

### CODE REVIEW

File: server/src/main/java/com/cloud/storage/StorageManagerImpl.java [hunk 1]
```diff
@@ -2075,6 +2075,7 @@ public CapacityVO getStoragePoolUsedStats(Long poolId, Long clusterId, Long podI
         if (poolId != null) {
             sc.addAnd("hostOrPoolId", SearchCriteria.Op.EQ, poolId);
         }
+        sc.addAnd("parent", SearchCriteria.Op.EQ, 0);
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/storage/StorageManagerImpl.java:2078]: @DK101010 `parent` is of Long type in `StoragePoolVO`
```suggestion
        sc.addAnd("parent", SearchCriteria.Op.EQ, 0L);
```

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@harikrishna-patnala can you please review since you've worked with datastore clusters?

**Comment by rhtyd** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]

**Review by harikrishna-patnala — APPROVED** [review]
Thanks, @DK101010. We need to skip either of datastore cluster parent or its children.

LGTM

**Review by shwstppr — APPROVED** [review]
LGTM

**Pull request merged**